### PR TITLE
Add SignUp API call

### PR DESCRIPTION
**Pull request by Francois389**
Add the call to the API to create an account (a Salesman).

### CODE REVIEW

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java [hunk 1]
```diff
@@ -74,122 +84,130 @@ public void onClick(View v) {
...
+            sendInformationToSignInUser(firstNameText, lastNameText, latitudeValue, longitudeValue, mailText, passwordText);
         }
     }
 
     /**
      * Check the first name field.
+     *
      * @return errorMessage
      */
-    public String checkFirstName(String firstNameText){
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java:88]: Change all checks to another util class

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import fr.iut_rodez.pathpilot_android_client.util.VolleyErrorHandler;
+
+public class SignUpService {
+
+    private static final String LOGIN_URL = BuildConfig.API_BASE_URL + "auth/signup";
+
+    private static final String TAG = SignUpService.class.getSimpleName();
+    public static final String CLE_MAIL = "mail";
+
+    public static void signUp(SignUpInput signUpInput, Context context) {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java:30]: Add signature to this method

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    Log.e(TAG, "Error while sending request", error);
+                    progressDialog.dismiss();
+                    VolleyErrorHandler.handleError(context, error);
+                }
+        );
+
+        requestQueue.add(request);
+    }
+
+    public static class SignUpInput {
```
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java:73]: Add signature there also

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/Network.java
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/Network.java:1]: isNetworkConnectedNeeded then ?

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/Popup.java
Comment (Odonata971) [inline app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/Popup.java:1]: I like it very much!! Good job

**Review by Odonata971 — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java (hunk 3 above)
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java:89]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/43/commits/13651282009dcc58f629be9db6b86f229a8c799b

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/Network.java
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/Network.java:1]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/43/commits/f687f50ffdce9284ba71adb3d92200ddec76c3e3

File: app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java [hunk 4]
```diff
@@ -1,16 +1,31 @@
...
 import androidx.activity.EdgeToEdge;
 import androidx.appcompat.app.AppCompatActivity;
 
-public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+import fr.iut_rodez.pathpilot_android_client.R;
+import fr.iut_rodez.pathpilot_android_client.login.Login;
+import fr.iut_rodez.pathpilot_android_client.signup.SignUpService.SignUpInput;
+import fr.iut_rodez.pathpilot_android_client.util.Popup;
+import fr.iut_rodez.pathpilot_android_client.util.ValidateForm;
+
```
[the remark this answers is not in the document]
Comment (Francois389) [reply app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUp.java:24]: Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/43/commits/633cf00a8385bc20fd7d05b35426296806f120cb

**Review by Odonata971 — APPROVED** [review]
LGTM

**Pull request merged**